synapse/lib/layer.py

```diff
@@ -4093,6 +4093,89 @@ async def iterNodeDataKeys(self, buid):
             prop = self.getAbrvProp(abrv)
             yield prop[0]
 
+    async def iterLayerAddPerms(self):
+
+        # nodes & props
+        for byts, abrv in self.propabrv.slab.scanByFull(db=self.propabrv.name2abrv):
+            form, prop = s_msgpack.un(byts)
+            if form is None:
+                continue
+
+            for _ in self.layrslab.scanByPref(abrv, db=self.byform):
```

Comment:
You'll actually want to do this on the `byprop` db, and it only needs to be checked when `prop` is None so you could have it be the `else` for the next block

Reply:
(`byprop` with no prop specified is where the primary prop for a node is stored, the `byform` index is the new index that links a sode with anything in the layer to its for for migration helpers)